core/conversion/converters/impl/conv_deconv.cpp

```diff
@@ -14,15 +14,46 @@ bool add_conv_deconv(ConversionCtx* ctx, const torch::jit::Node* n, args& args)
   auto in = args[0].ITensor(); // assumes non-static input Tensor
   auto w = Weights(ctx, args[1].unwrapToTensor());
   auto stride = util::toDims(args[3].unwrapToIntList());
-  LOG_DEBUG("stride: " << stride);
   auto padding = util::toDims(args[4].unwrapToIntList());
-  LOG_DEBUG("padding: " << padding);
   auto dilation = util::toDims(args[5].unwrapToIntList());
-  LOG_DEBUG("dilation: " << dilation);
   bool transposed = args[6].unwrapToBool();
   auto out_padding = util::toDims(args[7].unwrapToIntList());
-  LOG_DEBUG("out_padding: " << out_padding);
   int64_t groups = args[8].unwrapToInt();
+
+  auto dims = in->getDimensions();
+  auto orig_dims = dims;
+  LOG_DEBUG("Original input dims: " << orig_dims);
+
+  // Expand spatial dims from 1D to 2D if needed
+  auto expandDims = addPaddingLayer(ctx, n, in, 4);
+  if (expandDims) {
+    auto tensorPtr = expandDims->getOutput(0);
+    assert(tensorPtr);
+    dims = tensorPtr->getDimensions();
+    in = tensorPtr;
+  }
+  if (w.shape.nbDims < 4) {
+    for (int i = w.shape.nbDims; i < 4; ++i)
+      w.shape.d[i] = 1;
+    w.shape.nbDims = 4;
+    w.kernel_shape.nbDims = 2;
+    w.kernel_shape.d[1] = 1;
+  }
+  if (stride.nbDims==1)
+    stride = util::unsqueezeDims(stride, 1, 1);
+  if (dilation.nbDims==1)
+    dilation = util::unsqueezeDims(dilation, 1, 1);
+  if (padding.nbDims==1)  
+    padding = util::unsqueezeDims(padding, 1, 0);
+  if (out_padding.nbDims==1)
+    out_padding = util::unsqueezeDims(out_padding, 1, 0);
+
```

Comment:
I think its worth having the logging for what the shapes are when pulled right out of the args structure and then adding additional logging that says we are unsqueezing to 2D. That way its explicit what is happening

Reply:
I know we do that already for the input dims but I think it would be nice for the rest too

Reply:
That I think would be too much logging, and not very useful.